### DISCUSSION TITLE
Impossible upgrade from 1.14 to 1.15 Error when run kimai:update

### DISCUSSION THREAD

**Discussion by abkrim** [kimai/kimai#3242] [On-Premise: Installation & Updates]
I've error on update from version 1.14 to 1.15

I run without poblems from 1.13 to 1.14

```bash
Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install [OK]

kimai@central:~/web/timetracker.castris.com$ bin/console kimai:update

Kimai updates running ...
=========================



 [OK] Already at the latest version ("DoctrineMigrations\Version20210802174320")



 Rebuilding your cache, please be patient ...

 // Clearing the cache for the prod environment with debug false

09:22:07 CRITICAL  [php] Uncaught Error: syntax error, unexpected token "readonly" ["exception" => ParseError { …}]

In AnnotationDriver.php line 109:

  syntax error, unexpected token "readonly"

```

I've tried with shell with php 7.4 and php 8.0

Same error.

**Comment by kevinpapst** [comment]
Likely this here: https://github.com/schmittjoh/serializer/pull/1333

So you are running PHP 8.1, which is definitely not compatible with these old Kimai versions.

**Comment by abkrim** [comment]
No.

After furts problme, siwthc to 74. for upgrade to 1.13
After read docs, I see  that I can use 8.0 for 1.14, I  change to 8.0 in shell user. Double verified.

Errors is in 1.14 with php8.0 on shell.

Your UPGRADE text say can I use 8.0

Best regards.
